src/main/resources/static/api.yml

```diff
@@ -0,0 +1,616 @@
+        period:
+          type: integer
+          format: int64
+          example: 5
+          description: Duration of the treatment in days
+        datePrescribed:
+          type: string
+          format: date-time
+          description: Date and time of prescription
+    NewEvent:
+      type: object
+      properties:
+        prescriptionId:
+          type: string
+          example: 123e4567-e89b-12d3-a456-426614174223
+          description: Related prescription's id in Prescriptions table
+        dateTime:
+          type: string
+          format: date-time
+          description: Scheduled date and time for the event.
+        status:
+          type: string
+          description: Current event status
+          example: Done
+          enum:
+            - Scheduled
+            - Done
+    Event:
+      type: object
+      properties:
+        id:
+          type: string
+          example: 123e4567-e89b-12d3-a456-426614174233
+          description: Unique event id
+        prescriptionId:
+          type: string
+          example: 123e4567-e89b-12d3-a456-426614174223
+          description: Related prescription's id in Prescriptions table
+        patientId:
+          type: string
+          example: 123e4567-e89b-12d3-a456-426614174342
+          description: Related patent's id in Patient table
+        dateTime:
+          type: string
+          format: date-time
+          description: Scheduled date and time for the event.
+        status:
+          type: string
+          description: Current event status
+          example: Cancelled
+          enum:
+            - Scheduled
+            - Done
+            - Cancelled
+        cancelReason:
+          type: string
+          description: Reason for cancelling the event
+          example: Doctor's orders
+          default: n/a
+    Treatment:
+      type: object
+      properties:
+        id:
+          type: string
+          example: 123e4567-e89b-12d3-a456-4266141742334
+          description: Unique treatment identifier in catalogue
+        name:
+          type: string
+          example: Colonoscopy
+          description: Name of procedure/medicine
+        type:
+          type: string
+          description: Whether the treatment is a procedure or a medicine
+          example: Procedure
+          enum:
+            - Procedure
+            - Medicine
+    NewUser:
+      type: object
+      properties:
+        firstName:
+          type: string
+          example: John
+          description: The user's first name
+        lastName:
+          type: string
+          example: Doe
+          description: The user's last name
+        userName:
+          type: string
+          example: aolvera
+          description: The user's login identifier
+        password:
+          type: string
+          format: password
+          example: ao12!B45
+          description: The user's password
+        role:
+          type: array
```

Comment:
user might have the only one role, it should be String value but not the array

Reply:
Ok, saw that strategy on an article, for several roles per user... Does make sense in our case to keep just one per user. Changed the way roles are handled in security as well, now using single string value for roles this commit.